File: javascript-closure-angular/API/Client/RequestTimeFilter.js
goog.provide('API.Client.RequestTimeFilter');

/**
 * @record
 */
API.Client.RequestTimeFilter = function() {}

/**
 * @type {!Array<!API.Client.RequestLocation>}
 * @export
 */
API.Client.RequestTimeFilter.prototype.locations;

/**
 * @type {!Array<!API.Client.RequestTimeFilterDepartureSearch>}
 * @export
 */
API.Client.RequestTimeFilter.prototype.departureSearches;

/**
 * @type {!Array<!API.Client.RequestTimeFilterArrivalSearch>}
 * @export
 */
API.Client.RequestTimeFilter.prototype.arrivalSearches;

File: scalaz/src/main/scala/org/openapitools/client/api/RequestTimeFilter.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime
import RequestTimeFilter._

case class RequestTimeFilter (
  locations: List[RequestLocation],
departureSearches: Option[List[RequestTimeFilterDepartureSearch]],
arrivalSearches: Option[List[RequestTimeFilterArrivalSearch]])

object RequestTimeFilter {
  import DateTimeCodecs._

  implicit val RequestTimeFilterCodecJson: CodecJson[RequestTimeFilter] = CodecJson.derive[RequestTimeFilter]
  implicit val RequestTimeFilterDecoder: EntityDecoder[RequestTimeFilter] = jsonOf[RequestTimeFilter]
  implicit val RequestTimeFilterEncoder: EntityEncoder[RequestTimeFilter] = jsonEncoderOf[RequestTimeFilter]
}
